src/NServiceBus.AcceptanceTests/Sagas/When_sagaV2.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_sagaV2 : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_execute_saga()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<SagaEndpoint>(b => b
+                    .When(session => session.SendLocal(new StartSagaMessage
+                    {
+                        Key = "Part1_Part2"
+                    })))
+                .Done(c => c.SecondMessageReceived)
+                .Run();
+
+            Assert.IsTrue(context.SecondMessageReceived);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FirstMessageReceived { get; set; }
+            public bool SecondMessageReceived { get; set; }
+            public Guid SagaId { get; set; }
+        }
+
+        public class SagaEndpoint : EndpointConfigurationBuilder
+        {
+            public SagaEndpoint()
+            {
+                EndpointSetup<DefaultServer>();
+            }
+
+            public class ASagaV2 : SagaV2<ASagaV2.SagaV2Data>,
+                IAmStartedByMessages<StartSagaMessage>,
+                IAmStartedByMessages<OtherMessage>
+            {
+
+                Context context;
+
+                public ASagaV2(Context context)
+                {
+                    this.context = context;
+                }
+
+                protected override string CorrelationPropertyName => nameof(SagaV2Data.CorrelationId);
```

Comment:
Would it work to return here an expression that selects the property from the saga data?